src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -981,11 +981,24 @@ void LedgerConsensusImp::accept (std::shared_ptr<SHAMap> set)
         auto buildLCL = std::make_shared<Ledger>(
             *mPreviousLedger,
             app_.timeKeeper().closeTime());
-        auto constexpr v2_ledger_seq_switch = 40'000'000;
-        if (buildLCL->info().seq > v2_ledger_seq_switch &&
-           !buildLCL->stateMap().is_v2())
+        auto const v2_enabled = buildLCL->rules().enabled(featureSHAMapV2,
+                                                       app_.config().features);
+        auto v2_transition = false;
+        if (v2_enabled && !buildLCL->stateMap().is_v2())
         {
-            buildLCL->make_v2();
+            if (mHaveCorrectLCL)
+            {
+                int txCount = 0;
+                for (auto& txn : *set)
```

Comment:
There's no reason to clean up code that exists only to reproduce a bug.

DS
On Jun 23, 2016 7:59 PM, "Howard Hinnant" <notifications@github.com> wrote:

> In src/ripple/app/ledger/impl/LedgerConsensusImp.cpp
> <https://github.com/ripple/rippled/pull/1737#discussion_r68348059>:
>
> >          {
> > -            buildLCL->make_v2();
> > +            if (mHaveCorrectLCL)
> > +            {
> > +                int txCount = 0;
> > +                for (auto& txn : *set)
>
> Interesting. Yeah, I'll fix that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ripple/rippled/pull/1737/files/2f8a5ebe5211ab24cfa1ec79ee8e5d65e63ee95a#r68348059>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1j7QfkX_kqg9EfsQ-t9Xl1sDljzwL9ks5qO0f0gaJpZM4IvFh3>
> .
>